android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -694,45 +719,77 @@ public void onSettingsClicked() {
+                if (SharedPreferencesManager.getInstance().readBoolean(
+                            PlaylistPreferenceUtils.SHOULD_SHOW_PLAYLIST_ONBOARDING, true)) {
+                    View playlistButton = viewGroup.findViewById(R.id.playlist_button_id);
+                    if (playlistButton != null) {
+                        playlistButton.post(new Runnable() {
+                            @Override
+                            public void run() {
+                                PlaylistOnboardingActionClickListener
+                                        playlistOnboardingActionClickListener =
+                                                new PlaylistOnboardingActionClickListener() {
+                                                    @Override
+                                                    public void onOnboardingActionClick() {
+                                                        // addMediaToPlaylist(contentUrl,
+                                                        // viewGroup);
+                                                        addMediaToPlaylist(items);
+                                                    }
+                                                };
+                                try {
+                                    new PlaylistOnboardingPanel(
+                                            (FragmentActivity) BraveActivity.getBraveActivity(),
+                                            playlistButton, playlistOnboardingActionClickListener);
+                                } catch (BraveActivity.BraveActivityNotFoundException e) {
+                                    Log.e(TAG, "showPlaylistButton " + e);
+                                }
+                            }
+                        });
+                    }
+                    SharedPreferencesManager.getInstance().writeBoolean(
+                            PlaylistPreferenceUtils.SHOULD_SHOW_PLAYLIST_ONBOARDING, false);
+                }
             }
-
         } catch (BraveActivity.BraveActivityNotFoundException e) {
             Log.e(TAG, "showPlaylistButton " + e);
         }
     }
 
-    private void addMediaToPlaylist(org.chromium.url.mojom.Url contentUrl, ViewGroup viewGroup) {
+    private void addMediaToPlaylist(PlaylistItem[] items) {
         if (mPlaylistService == null) {
             return;
         }
-        boolean shouldCacheOnlyOnWifi =
-                (SharedPreferencesManager.getInstance().readInt(
-                         BravePlaylistPreferences.PREF_AUTO_SAVE_MEDIA_FOR_OFFLINE, 0)
-                                == 2
-                        && ConnectionUtils.isWifiAvailable(getContext()));
+        mPlaylistService.addMediaFiles(
+                items, ConstantUtils.DEFAULT_PLAYLIST, shouldCacheMediaFilesForPlaylist());
+        int mediaCount = SharedPreferencesManager.getInstance().readInt(
+                PlaylistPreferenceUtils.ADD_MEDIA_COUNT);
+        if (mediaCount < 3) {
```

Comment:
It's hard to understand what is this number about, maybe worth it to define as a const with meaningful name.